tools/docs/generate_version_histories.py

```diff
@@ -0,0 +1,265 @@
+        return pathlib.Path(self.args.version_path)
+
+    def add_arguments(self, parser) -> None:
+        super().add_arguments(parser)
+        parser.add_argument("version_path")
+        parser.add_argument("output_file")
+        parser.add_argument("changelogs", nargs="+")
+
+    def minor_index_path(self, minor_version) -> pathlib.Path:
+        return self.tpath.joinpath(f"v{minor_version.base_version}").joinpath(
+            f"v{minor_version.base_version}.rst")
+
+    @runner.cleansup
+    async def run(self) -> None:
+        self.write_version_history_index()
+        self.write_version_histories()
+        self.write_version_history_minor_indeces()
+        self.write_tarball()
+
+    def write_tarball(self) -> None:
+        with tarfile.open(self.args.output_file, "w") as tarball:
+            tarball.add(self.tpath, arcname="./")
+
+    def write_version_histories(self) -> None:
+        for changelog_version in self.project.changelogs:
+            self.write_version_history(changelog_version)
+
+    def write_version_history(self, changelog_version: version.Version) -> None:
+        minor_version = minor_version_for(changelog_version)
+        root_path = self.tpath.joinpath(f"v{minor_version.base_version}")
+        root_path.mkdir(parents=True, exist_ok=True)
+        version_history = self.version_history_tpl.render(
+            current_minor_version=self.project.current_minor_version,
+            minor_version=minor_version,
+            current_version=self.project.current_version,
+            sections=self.sections,
+            changelog=self.project.changelogs[changelog_version])
+        version_path = root_path.joinpath(f"v{changelog_version}.rst")
+        version_path.write_text(f"{version_history.strip()}\n\n")
+
+    def write_version_history_index(self) -> None:
+        stable_message = (
+            "Versions that are currently supported." if self.project.is_main_dev else
+            "Versions that were supported when this branch was initially released.")
+        archived_message = (
+            "Versions that are no longer supported." if self.project.is_main_dev else
+            "Versions that were no longer supported when this branch was initially released.")
+        version_history_rst = self.version_history_index_tpl.render(
+            archived_message=archived_message,
+            stable_message=stable_message,
+            dev_version=self.project.dev_version,
+            changelogs=self.project.changelogs,
+            minor_versions=self.project.minor_versions,
+            stable_versions=self.project.stable_versions,
+            archived_versions=self.project.archived_versions)
+        self.tpath.joinpath("version_history.rst").write_text(f"{version_history_rst.strip()}\n\n")
+
+    def write_version_history_minor_indeces(self) -> None:
+        for i, (minor_version, patches) in enumerate(self.project.minor_versions.items()):
+            if self.project.is_main_dev and i == 0:
```

Comment:
probs this should be any dev version

Reply:
actualy this is correct - only main dev means that the minor version is not added